air/src/execution_step/execution_context/context.rs

```diff
@@ -193,8 +193,12 @@ impl ExecutionCtx<'_> {
             None
         };
 
-        self.error_descriptor
-            .try_to_set_error_from_exec_error(error, instruction, peer_id, tetraplet.clone());
+        if self.error_descriptor.error_setting_is_enabled() {
```

Comment:
it's better to move this check inside the `try_to_set_error_from_exec_error` function.

Reply:
Done